pos_lot_selection/static/src/js/popup/ProductScreen.js

```diff
@@ -0,0 +1,57 @@
+/*
+    Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
+    License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+*/
+odoo.define("pos_lot_selection.ProductScreen", function (require) {
+    "use strict";
+
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const Registries = require("point_of_sale.Registries");
+
+
+    const PosLotSaleProductScreen = (ProductScreen) =>
+        class extends ProductScreen {
+            async _getAddProductOptions(product, base_code) {
+                if (["serial", "lot"].includes(product.tracking)) {
+                    // Set lots to context as it one possible way to extract arguments
+                    // later in popup
+                    this.env.session.lots = await this.get_lots(product);
+                } else {
+                    // Cleanup
+                    this.env.session.product = false;
+                }
+
+                return super._getAddProductOptions(product, (base_code = base_code));
+            }
+
+            async get_lots(product) {
+                var lots = await this.rpc(
+                    {
+                        model: "stock.production.lot",
+                        method: "search_read",
+                        kwargs: {
+                            domain: [
+                                "&",
+                                "&",
+                                ["product_id", "=", product.id],
+                                ["available_on_pos", "=", true],
```

Comment:
```suggestion
                                ["product_qty", ">", 0],
```

Reply:
product_qty is not a stored field so we need to use a filter here, having a boolean field looks like more explicit way